pkg/cmd/tracee.go

```diff
@@ -88,38 +89,44 @@ func (r Runner) Run(ctx context.Context) error {
 		}
 	}()
 
-	stream := t.SubscribeAll()
-	defer t.Unsubscribe(stream)
+	var stream *streams.Stream
 
-	// Preeamble
+	if !r.Printer.Ignore() {
+		stream = t.SubscribeAll()
+		defer t.Unsubscribe(stream)
 
-	r.Printer.Preamble()
+		// Preamble
 
-	// Start event channel reception
+		r.Printer.Preamble()
 
-	go func() {
-		for {
-			select {
-			case event := <-stream.ReceiveEvents():
-				r.Printer.Print(event)
-			case <-ctx.Done():
-				return
+		// Start event channel reception
+
+		go func() {
+			for {
+				select {
+				case event := <-stream.ReceiveEvents():
+					r.Printer.Print(event)
+				case <-ctx.Done():
+					return
+				}
 			}
-		}
-	}()
+		}()
+	}
 
 	// Blocks (until ctx is Done)
 	err = t.Run(ctx)
 
-	// Drain remaininig channel events (sent during shutdown),
-	// the channel is closed by the tracee when it's done
-	for event := range stream.ReceiveEvents() {
-		r.Printer.Print(event)
-	}
+	if !r.Printer.Ignore() {
+		// Drain remaining channel events (sent during shutdown),
+		// the channel is closed by the tracee when it's done
+		for event := range stream.ReceiveEvents() {
+			r.Printer.Print(event)
+		}
 
-	stats := t.Stats()
-	r.Printer.Epilogue(*stats)
-	r.Printer.Close()
+		stats := t.Stats()
+		r.Printer.Epilogue(*stats)
+		r.Printer.Close()
+	}
 
 	return err
```

Comment:
Instead of adding the extra indentation, you can add before this code block:
```
if r.Printer.Ignore() {
	// Blocks (until ctx is Done)
	err = t.Run(ctx)
}
```

Reply:
`t.Run(ctx)` always need to be called so we could wrap the logic of running with Printer like:

```
if r.Printer.Ignore() {
	// Just run the tracee with no printer handling,
    // blocking (until ctx is Done)
	return t.Run(ctx)
}

err = runWithPrinter(ctx, r, t) // this will also call t.Run(ctx)
```

WDYT?

Reply:
Done.